pkg/strategy/dca2/maker.go

```diff
@@ -0,0 +1,108 @@
+package dca2
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func (s *Strategy) placeMakerOrders(ctx context.Context) error {
+	s.logger.Infof("[DCA] start placing maker orders")
+	price, err := s.getBestPriceUntilSuccess(ctx, s.Short)
+	if err != nil {
+		return err
+	}
+
+	orders, err := s.generateMakerOrder(s.Short, s.Budget, price, s.PriceDeviation, s.MaxOrderNum)
+	if err != nil {
+		return err
+	}
+
+	createdOrders, err := s.OrderExecutor.SubmitOrders(ctx, orders...)
+	if err != nil {
+		return err
+	}
+
+	s.debugOrders(createdOrders)
+
+	return nil
+}
+
+func (s *Strategy) getBestPriceUntilSuccess(ctx context.Context, short bool) (fixedpoint.Value, error) {
```

Comment:
I think you are combining two behavior in this function

1. make one function calls QueryTicker func with backoff
2. wrap the 1. func with the short option to get the price by side